packages/modeling/src/geometries/geom2/fromSides.js

```diff
@@ -0,0 +1,118 @@
+import * as vec2 from '../../maths/vec2/index.js'
+import create from './create.js'
+
+/*
+ * Create a list of edges which SHARE vertices.
+ * This allows the edges to be traversed in order.
+ */
+const toSharedVertices = (sides) => {
+  const unique = new Map() // {key: vertex}
+  const getUniqueVertex = (vertex) => {
+    const key = vertex.toString()
+    if (unique.has(key)) {
+      return unique.get(key)
+    } else {
+      unique.set(key, vertex)
+      return vertex
+    }
+  }
+
+  return sides.map((side) => side.map(getUniqueVertex))
+}
+
+/*
+ * Convert a list of sides into a map from vertex to edges.
+ */
+const toVertexMap = (sides) => {
+  const vertexMap = new Map()
+  // first map to edges with shared vertices
+  const edges = toSharedVertices(sides)
+  // construct adjacent edges map
+  edges.forEach((edge) => {
+    if (vertexMap.has(edge[0])) {
+      vertexMap.get(edge[0]).push(edge)
+    } else {
+      vertexMap.set(edge[0], [edge])
+    }
+  })
+  return vertexMap
+}
+
+/**
+ * Create the outline(s) of the given geometry.
+ * @param {geom2} geometry - geometry to create outlines from
```

Comment:
param is sides

returns geom2

I guess this is being used internally by the booleans. is this still necessary?

Reply:
geom2.fromSides is still needed in 7 places, for expansion and extrusion. My goal is to remove these uses, but it will require refactoring, and this PR was already long enough. I could remove it from the index, so that it is internal use only?